core/src/main/java/net/opentsdb/query/processor/groupby/GroupBy.java

```diff
@@ -63,7 +77,30 @@ public GroupBy(final QueryNodeFactory factory,
     }
     this.config = config;
   }
-    
+
+  @Override
+  public Deferred<Void> initialize(Span span) {
+    return super.initialize(span)
+        .addCallback(
+            arg -> {
+              for (QueryNode node : (Collection<QueryNode>) this.downstream) {
+                if (node instanceof TimeSeriesDataSource) {
+                  TimeSeriesDataSource timeSeriesDataSource = (TimeSeriesDataSource) node;
+                  TimeSeriesDataSourceConfig config = (TimeSeriesDataSourceConfig) timeSeriesDataSource.config();
+                  List<QueryNodeConfig> pushDownNodes = config.getPushDownNodes();
+                  for (QueryNodeConfig queryNodeConfig : pushDownNodes) {
+                      if(queryNodeConfig instanceof DownsampleConfig) {
+                        downsampleConfig = (DownsampleConfig) queryNodeConfig;
+                        break;
+                      }
+                  }
+                  break;
+                }
```

Comment:
For the HBase case, also check to see if the node is an instance of `Downsample` and grab the config from there.